components/scheduler/src/Scheduler.svelte

```diff
@@ -1,24 +1,94 @@
 <svelte:options tag="nylas-scheduler" />
 
 <script lang="ts">
-  import { ManifestStore } from "../../../commons/src";
+  import {
+    ManifestStore,
+    EventStore,
+    AvailabilityStore,
+  } from "../../../commons/src";
+  import { createEvent } from "@commons/connections/events";
+
   import { onMount, tick } from "svelte";
   import "../../availability/src/Availability.svelte";
 
-  export let id: string = "";
-  export let access_token: string = "";
+  import {
+    getPropertyValue,
+    buildInternalProps,
+  } from "@commons/methods/component";
 
   import type { Manifest } from "@commons/types/Scheduler";
+  import type { TimeSlot } from "@commons/types/Availability";
+  import type { TimespanEvent, EventQuery } from "@commons/types/Events";
 
+  // #region props
+  export let id: string = "";
+  export let access_token: string = "";
+  export let availability_id: string;
+  export let email_ids: string[];
+  // #endregion props
+
+  //#region mount and prop initialization
+  let internalProps: Partial<Manifest> = {};
   let manifest: Partial<Manifest> = {};
   onMount(async () => {
     await tick();
     const storeKey = JSON.stringify({ component_id: id, access_token });
     manifest = (await $ManifestStore[storeKey]) || {};
+
+    internalProps = buildInternalProps($$props, manifest) as Partial<Manifest>;
   });
 
-  function timeSlotChosen({ detail }) {
-    console.log("timeslots chosen", detail.timeSlots);
+  $: {
+    const rebuiltProps = buildInternalProps(
+      $$props,
+      manifest,
+    ) as Partial<Manifest>;
+    if (JSON.stringify(rebuiltProps) !== JSON.stringify(internalProps)) {
+      internalProps = rebuiltProps;
+    }
+  }
+
+  $: {
+    availability_id = getPropertyValue(
+      internalProps.availability_id,
+      availability_id,
+      "",
+    );
+    email_ids = getPropertyValue(internalProps.email_ids, email_ids, []);
+  }
+  // #endregion mount and prop initialization
+
+  let timeSlots: TimeSlot[] = [];
+
+  function timeSlotChosen({ detail }: CustomEvent) {
+    timeSlots = detail.timeSlots;
+  }
+
+  async function bookTimeSlots(events: TimeSlot[]) {
+    const bookings = events.map(async (event) => {
+      let postableEvent: Partial<TimespanEvent> = {
+        title: "My test event",
+        participants: [], // TODO: add to the participants array from availbility.event.available_calendars
+        calendar_id: event.calendar_id,
+        when: {
+          start_time: event.start_time.getTime() / 1000,
+          end_time: event.end_time.getTime() / 1000,
+        },
+      };
+      return await createEvent(
```

Comment:
Technically you should not need to await this promise here and simply return it.  Right now the code is awaiting for the results, wrapping it into a promise again, which is awaited for again.